deepomatic/http_helper.py

```diff
@@ -114,46 +114,50 @@ def format_params(self, data):
                     data[key] = json.dumps(value)
         return data
 
-    def dump_json_for_multipart(self, data, files):
-        def recursive_json_dump(prefix, obj, data, files, omit_dot=False):
+    def dump_json_for_multipart(self, dic):
+        if dic is None:
+            return None
+
+        def recursive_json_dump(prefix, obj, dic, omit_dot=False):
             if isinstance(obj, dict):
                 if not omit_dot:  # see comment below
                     prefix += '.'
                 for key, value in obj.items():
-                    recursive_json_dump(prefix + key, value, data, files)
+                    recursive_json_dump(prefix + key, value, dic)
             elif isinstance(obj, list):
                 for i, value in enumerate(obj):
                     # omit_dot is True as DRF parses list of dictionnaries like this:
                     # {"parent": [{"subfield": 0}]} would be:
                     # 'parent[0]subfield': 0
-                    recursive_json_dump(prefix + '[{}]'.format(i), value, data, files, omit_dot=True)
-            elif hasattr(obj, 'read'):  # if is file:
-                if prefix in files:
-                    raise DeepomaticException("Duplicate key: " + prefix)
-                files[prefix] = obj
+                    recursive_json_dump(prefix + '[{}]'.format(i), value, dic, omit_dot=True)
             else:
-                data[prefix] = obj
+                if prefix in new_dic:
+                    raise DeepomaticException("Duplicate key: " + prefix)
+                dic[prefix] = obj
 
-        if files is None:
-            files = {}
-        new_data = {}
+        new_dic = {}
 
-        recursive_json_dump('', data, new_data, files, omit_dot=True)
+        recursive_json_dump('', dic, new_dic, omit_dot=True)
 
-        if len(files) == 0:
-            files = None
-        return new_data, files
+        return new_dic
 
     def make_request(self, func, resource, params=None, data=None, content_type='application/json', files=None, stream=False, *args, **kwargs):
-        if isinstance(data, dict) or isinstance(data, list):
-            if content_type is not None:
-                if content_type.strip() == 'application/json':
+
+        if content_type is not None:
+            if content_type.strip() == 'application/json':
+                if data is not None:
                     data = json.dumps(data)
-                elif content_type.strip() == 'multipart/mixed':
-                    content_type = None  # will be automatically set to multipart
-                    data, files = self.dump_json_for_multipart(data, files)
-                else:
-                    raise DeepomaticException("Unsupported Content-Type")
+            elif content_type.strip() == 'multipart/mixed':
+                # If no files are provided, requests will default to form-urlencoded content type
+                # But the API doesn't support it.
+                if not files:
```

Comment:
So this means we cannot read frames from a camera and send them as binary ?

Reply:
As discussed it is still possible ;)